compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -77,8 +77,8 @@ pub fn overlapping_impls(
     // a quick check via fast_reject to tell if the impl headers could possibly
     // unify.
     let drcx = DeepRejectCtxt { treat_obligation_params: TreatParams::AsInfer };
-    let impl1_ref = tcx.impl_trait_ref(impl1_def_id);
-    let impl2_ref = tcx.impl_trait_ref(impl2_def_id);
+    let impl1_ref = tcx.bound_impl_trait_ref(impl1_def_id).map(ty::EarlyBinder::subst_identity);
+    let impl2_ref = tcx.bound_impl_trait_ref(impl2_def_id).map(ty::EarlyBinder::subst_identity);
```

Comment:
there we actually pretend to instantiate both sides with inference variables so `skip_binder` is more correct here.

Note that the two impls can have different generic parameters so both can't be in the identity context. the `DeepRejectCtxt` is setup in a way to deal with `ty::Param` which are actually inference variables (but aren't for perf reasons), so it's fine for it to look into types even with `skip_binder`